tools/SourceKit/tools/sourcekitd/lib/API/sourcekitdAPI-InProc.cpp

```diff
@@ -0,0 +1,941 @@
+  llvm::report_fatal_error("invalid sourcekitd error object");
+}
+
+const char *
+sourcekitd_response_error_get_description(sourcekitd_response_t obj) {
+  if (SKDError *Error = dyn_cast<SKDError>(static_cast<SKDObject *>(obj)))
+    return Error->getDescription().data();
+  llvm::report_fatal_error("invalid sourcekitd error object");
+}
+
+static sourcekitd_variant_t variantFromSKDObject(SKDObjectRef Object);
+
+sourcekitd_variant_t
+sourcekitd_response_get_value(sourcekitd_response_t resp) {
+  if (sourcekitd_response_is_error(resp))
+    return makeNullVariant();
+  return variantFromSKDObject(static_cast<SKDObject *>(resp));
+}
+
+//===----------------------------------------------------------------------===//
+// Internal Request API
+//===----------------------------------------------------------------------===//
+
+void sourcekitd::printRequestObject(sourcekitd_object_t Obj, raw_ostream &OS) {
+  if (!Obj) {
+    OS << "<<NULL>>";
+    return;
+  }
+
+  SKDObjectPrinter(OS).visit(Obj);
+}
+
+//===----------------------------------------------------------------------===//
+// Internal ResponseBuilder Implementation
+//===----------------------------------------------------------------------===//
+
+ResponseBuilder::ResponseBuilder() {
+  Impl = retained(new SKDDictionary());
+}
+
+ResponseBuilder::~ResponseBuilder() {
+  static_cast<SKDObject *>(Impl)->Release();
+}
+
+ResponseBuilder::ResponseBuilder(const ResponseBuilder &Other) {
+  Impl = retained(static_cast<SKDObject *>(Other.Impl));
+}
+
+ResponseBuilder &ResponseBuilder::operator =(const ResponseBuilder &Other) {
+  auto Tmp = retained(static_cast<SKDObject *>(Other.Impl));
+  static_cast<SKDObject *>(Impl)->Release();
+  Impl = Tmp;
+  return *this;
+}
+
+ResponseBuilder::Dictionary ResponseBuilder::getDictionary() {
+  return Dictionary(Impl);
+}
+
+sourcekitd_response_t ResponseBuilder::createResponse() {
+  return retained(static_cast<SKDObject *>(Impl));
+}
+
+void ResponseBuilder::Dictionary::set(UIdent Key, SourceKit::UIdent UID) {
+  set(Key, SKDUIDFromUIdent(UID));
+}
+
+void ResponseBuilder::Dictionary::set(UIdent Key, sourcekitd_uid_t UID) {
+  static_cast<SKDObject *>(Impl)->set(SKDUIDFromUIdent(Key), new SKDUID(UID));
+}
+
+void ResponseBuilder::Dictionary::set(UIdent Key, const char *Str) {
+  set(Key, StringRef(Str));
+}
+
+void ResponseBuilder::Dictionary::set(UIdent Key, StringRef Str) {
+  static_cast<SKDObject *>(Impl)->set(SKDUIDFromUIdent(Key), 
+                                      new SKDString(Str.str()));
+}
+
+void ResponseBuilder::Dictionary::set(UIdent Key, int64_t Val) {
+  static_cast<SKDObject *>(Impl)->set(SKDUIDFromUIdent(Key), new SKDInt64(Val));
+}
+
+void ResponseBuilder::Dictionary::set(SourceKit::UIdent Key,
+                                      ArrayRef<StringRef> Strs) {
+  auto ArrayObject = new SKDArray();
+  for (auto Str : Strs) {
+    ArrayObject->set(SOURCEKITD_ARRAY_APPEND, new SKDString(Str.str()));
```

Comment:
You don't generally need to write `<StringRef>.str()`, `StringRef` has conversion operator to `std::string`, I believe `new SKDString(Str)` will work.